packages/lsp/src/connection_manager.ts

```diff
@@ -198,6 +247,16 @@ export class DocumentConnectionManager
     this.disconnectDocumentSignals(foreignDocument);
   }
 
+  /**
+   * Update the active adapter.
+   *
+   * @param  path - path to the inner document of the adapter or null.
+   */
+  updateCurrentAdapter(path: string | null): void {
+    this._currentAdapter = path;
+    this._currentAdapterChanged.emit(path);
+  }
```

Comment:
Same problem as before. In addition, here, we are emitting `currentAdapterChanged` before the adapter is registered. When a plugin subscribes to this signal, the plugin needs to check whether the adapter exist or wait for `adapterAdded` signal.

A possible solution could be to debounce emitting `currentAdapterChanged` until the adapter exists.